randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -25,7 +44,45 @@ def get_item_id_for_item(item: ItemResourceInfo) -> str:
         raise KeyError(f"{item.long_name} has no item ID.") from e
 
 
+def convert_conditional_resource(res: ConditionalResources) -> Iterator[dict]:
+    if not res.resources:
+        yield {"item_id": "ITEM_NONE", "quantity": 0}
+        return
+
+    for resource in reversed(res.resources):
+        item_id = get_item_id_for_item(resource[0])
```

Comment:
We could maybe cast this to remove the mypy error?

Reply:
`ConditionalResources.resources` should use `ItemResourceInfo` instead of `ResourceInfo`. Don't think it makes sense for a pickup to give non-item resources.

Reply:
Isn't this already using `ItemResourceInfo`?

Reply:
`ConditionalResources.resources` is currently of type `ResourceGainTuple` which is `tuple[ResourceQuantity, ...]` where `ResourceQuantity` is `tuple[ResourceInfo, int]` and `ResourceInfo` is `ResourceInfo = SimpleResourceInfo | ItemResourceInfo | TrickResourceInfo | NodeResourceInfo`

I'm also fine with doing this in a future pr if you don't feel comfortable in changing these things in randovania.

Reply:
Yeah I'll leave this then. I'll push what I have for now.